src/main/java/org/eclipse/pass/schema/SchemaServlet.java

```diff
@@ -15,27 +41,42 @@
 @WebServlet("/schemaservice")
 public class SchemaServlet extends HttpServlet {
     private static final long serialVersionUID = 1L;
+    private List<String> repository_list;
+    private String next;
+    private PassClient client;
 
     /**
      * Servlet constructor.
      */
     public SchemaServlet() {
-        // TODO: Not implemented yet
+        client = PassClientFactory.getPassClient();
+        repository_list = new ArrayList<String>();
     }
 
-    /**
-     * Handle GET requests by sending a HTML response back to the client, providing
-     * a link to the PASS schema service documentation.
-     */
-    @Override
-    protected void doGet(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-
-        // setCommonHeaders()
+    // used for unit testing to insert a mock client
+    public SchemaServlet(PassClient client) {
+        this.client = client;
+        repository_list = new ArrayList<String>();
+    }
 
-        // Create HTML response with a redirect link to the PASS schema service
-        // documentation
+    protected List<String> readText(BufferedReader r) throws IOException {
```

Comment:
There is some standard helper method to do this, but I can't remember it offhand.